### ISSUE TITLE
neo.compiler.csharp using dotnet 5.0 which is incompatible with mac m1 chip

### ISSUE THREAD

**Issue by RawichK** [neo-project/neo-devpack-dotnet#716]
Below is the console log when I tried to create smart contract using Neo Blockchain Toolkit for VS Code.
I search through Microsoft site already and found out that dotnet 5.0 for mac only got one option which is runtime for x64.
Therefore,  I would like to request a new version that support dotnet 6.0

Best regards,
Rawich Kiriyapong

> Executing task: dotnet tool restore <

Tool 'neo.express' (version '3.0.2') was restored. Available commands: neoxp
Tool 'neo.compiler.csharp' (version '3.0.0') was restored. Available commands: nccs

Restore was successful.

Terminal will be reused by tasks, press any key to close it.

> Executing task: dotnet build /property:GenerateFullPaths=true /consoleloggerparameters:NoSummary <

Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /Users/rawichk/projects/smart-contract/contracts/HelloWorld/src/HelloWorldContract.csproj (in 287 ms).
  Determining projects to restore...
  Restored /Users/rawichk/projects/smart-contract/contracts/HelloWorld/test/HelloWorldContractTests.csproj (in 361 ms).
  1 of 2 projects are up-to-date for restore.
  HelloWorldContract -> /Users/rawichk/projects/smart-contract/contracts/HelloWorld/src/bin/Debug/net5.0/HelloWorldContract.dll
  It was not possible to find any compatible framework version
  The framework 'Microsoft.AspNetCore.App', version '5.0.0' (arm64) was not found.
    - The following frameworks were found:
        6.0.1 at [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  
  You can resolve the problem by installing the specified framework and/or SDK.
  
  The specified framework can be found at:
    - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.AspNetCore.App&framework_version=5.0.0&arch=arm64&rid=osx.11.0-arm64

**Comment by devhawk** [comment]
You need to upgrade to 3.1 version of neo tools for .NET 6 support. Neo Express 3.1.38 uses .NET 6.  However, there is an issue getting neo express to find the rocksdb library. For now, the workaround is:

1. Install the arm64 version of dotnet 6
2. install rocksdb via homebrew
3. create a symbolic link to /opt/homebrew/lib/librocksdb.dylib in /usr/local/lib. I used this command: l`n -s /opt/homebrew/lib/librocksdb.dylib /usr/local/lib/librocksdb.dylib`

Next release of the rocksdb library that neo express uses will be able to [resolve librocksdb.dylib from the standard location homebrew installs it](https://github.com/curiosity-ai/rocksdb-sharp/issues/15#issuecomment-1016493056). At that point, the symbolic link step above will no longer be needed.

**Comment by RawichK** [comment]
Thank you for your help. But I'm already on 3.1 version by checking with this command
`
~ » dotnet tool list -g                                                                
Package Id               Version      Commands
----------------------------------------------
neo.compiler.csharp      3.1.0        nccs
neo.express              3.1.38       neoxp
`

Above error was display when I create smart contract using neo blockchain toolkit extension for vs code.
Please check screen shots below, I'm not sure why vscode target older version of neo tools. I would like your help to resolve these conflicts.
![image](https://user-images.githubusercontent.com/8307015/150239700-6a7e79c0-bbb4-48d7-aae7-ad16772156a7.png)
![image](https://user-images.githubusercontent.com/8307015/150239716-e62750b1-3ad3-428d-8230-48c2a5d063ca.png)

Best,

**Comment by RawichK** [comment]
After investigating for a while, I found the solution to this issue.
Generated project structure, configurations, and source code are target older version. 
That's why even I got latest tool version the compiler throw errors.

Resolving steps are as follow.
1. remove test directory to make it more simple.
2. update dirs.proj file, remove this line `<Projects Include="test/HelloWorldContractTests.csproj" />`
3. update dotnet-tools.json, update tools version, neo.express from 3.0.2 to `3.1.38`, neo.compiler.csharp from 3.0.0 to `3.1.0`
4. update src/HelloWorldContract.cs add using directive `using Neo.SmartContract.Framework.Attributes;` because  `ManifestExtra` is now part of this.
5. update src/HelloWorldContract.csproj, update target framework to `net6.0`, update package reference to Neo.SmartContract.Framework version `3.1.0`
6. Delete bin, obj directory then build again, that's it.

Hope these steps can help new joining developers that using mac m1.
